ls1intum/Artemis
# `Programming exercises`: Change templates for plain Java exercises to contain a build file

### Checklist
#### General
#### Server
#### Changes affecting Programming Exercises

### Motivation and Context
As discussed with @krusche, the project types PLAIN_MAVEN and PLAIN_GRADLE (for Java exercises) should contain a build configuration file to unify the import process into the local IDE and to make it independent of the used IDE. 

### Description
The changes to the templates are as following:
- PLAIN_MAVEN: Eclipse files have been removed and replaced with a pom.xml
- PLAIN_GRADLE: All required files for importing the exercise as a Gradle project have been added.
- Remove Eclipse-file-related functionality from the creation of new PLAIN_MAVEN exercises. This functionality will not be removed from the ProgrammingExerciseExportService to support exercises that were created with the eclipse project files.
- Client-side: The UI has changed for selecting the project type (see picture below).

**Node**: The project type enum values are not accurate anymore. However, this is not visible to the client and in consideration of including this PR into 5.7.4, this will be adjusted in a follow-up PR.

### Steps for Testing
Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration
3. Create new Java Programming Exercises with the project types PLAIN_MAVEN and PLAIN_GRADLE (only one configuration is required, SCA and sequential test runs are not affected by the changes).
4. Clone and import the template/solution repository into your local IDE as Maven/Gradle project and verify that everything works as expected.
5. Older PLAIN_MAVEN exercises that have been created when this PR is not deployed (with Eclipse project files) can be imported when this PR is deployed.
6. Create a programming exercise for every language/project type and verify that the exercise is created for the correct language/project type. The correct project type should be chosen no matter how you wildly click through the UI. This is cumbersome to test. I would suggest to clone locally, change the functionality of the "Generate"-Button to only alert about the chosen project type but not actually creating an exercise. If you need help, contact me.

### Screenshots
<img width="386" alt="Bildschirmfoto 2022-04-23 um 12 58 43" src="https://user-images.githubusercontent.com/73833780/164893664-338fb5df-4181-467f-b712-a84b6de6b750.png">
